spec/calendar.html

```diff
@@ -483,14 +483,36 @@ <h1>ISODaysInMonth ( _year_, _month_ )</h1>
       </emu-alg>
     </emu-clause>
 
-    <emu-clause id="sec-temporal-toisoweekofyear" aoid="ToISOWeekOfYear">
-      <h1>ToISOWeekOfYear ( _year_, _month_, _day_ )</h1>
+    <emu-clause id="sec-temporal-toisoweekofyear" type="abstract operation">
+      <h1>
+        ToISOWeekOfYear (
+          _year_: an integer,
+          _month_: an integer,
+          _day_: an integer,
+        ): an integer
+      </h1>
+      <dl class="header">
+        <dt>description</dt>
+        <dd>It implements the logic for the `Temporal.Calendar.prototype.weekOfYear` method, which is used for the ISO 8601 calendar.</dd>
+      </dl>
       <emu-alg>
-        1. Assert: _year_ is an integer.
-        1. Assert: _month_ is an integer.
-        1. Assert: _day_ is an integer.
-        1. Let _date_ be the date given by _year_, _month_, and _day_.
-        1. Return _date_'s week number according to ISO-8601 as an integer.
+        1. Let _dayOfYear_ be ToISODayOfYear(_year_, _month_, _day_).
+        1. Let _dayOfWeek_ be ToISODayOfWeek(_year_, _month_, _day_).
+        1. If _dayOfWeek_ is 0, then
+          1. Set _dayOfWeek_ to 7.
+        1. Let _week_ be floor((_dayOfYear_ - _dayOfWeek_ + 10) / 7).
+        1. If _week_ &lt; 1, then
+          1. Let _dayOfJan1st_ be ToISODayOfWeek(_year_, 1, 1).
+          1. If _dayOfJan1st_ is 5, then
+            1. Return 53.
+          1. If _dayOfJan1st_ is 6, then
+            1. Return 52 + ℝ(InLeapYear(TimeFromYear(𝔽(_year_ - 1)))).
+          1. Return 52.
+        1. If _week_ is 53, then
+          1. Let _test_ be InLeapYear(TimeFromYear(𝔽(_year_))) + 365 - _dayOfYear_ - 4 + _dayOfWeek_.
+          1. If _test_ &lt; 0, then
+            1. Return 1.
```

Comment:
these are a lot of magic numbers - can they be placed in named constants?

Reply:
I got them from [the](https://github.com/tc39/proposal-temporal/blob/main/polyfill/lib/ecmascript.mjs#L2387)
It is truly matic number to me. It will be nice if the polyfill author could suggest the name for such constant.  

@redneb  @ptomato @pipobscure

Reply:
See my comment at https://github.com/tc39/proposal-temporal/issues/1641#issuecomment-1216929403